api/google-manager/services/google-manager.js

```diff
@@ -110,8 +110,17 @@ class GoogleManager {
 
       return group.data;
     } catch(err) {
-      console.error(`Google Admin Directory API returned error ${err} when getting group`);
-      throw new Error(err);
+      if(err.code === 404) {
+        console.error('Google group does not exist');
+        /*
+          we have to return undefined to allow strapi's code
+          to work even if the project doesn't have a google group
+        */
+        return undefined;
```

Comment:
Do you mean instead of returning undefined to return an error message?

Reply:
If so, the reason I return undefined is so I can then check if getGroup returns undefined in extensions/content-manager/controllers/collection-types.js line 134. If it is undefined, I allow the rest of strapi's code to proceed to let them still be able to edit their project without getting an error trying to edit it. If I don't do this, it'll return an error when a user tries to update their project, and it won't let them edit their project. That's why I don't return an error.

Reply:
I see your reasoning. Is it idiomatic to return undefined to signal an error in javascript? If so I'm good with this change.